src/modules/verifiable-credentials/storage-client.ts

```diff
@@ -22,4 +32,80 @@ export default class VCStorageClient {
 
     return data;
   }
+
+  async store(
+    credentials: VerifiableCredential<CredentialSubject>[]
+  ): Promise<StoreVcResult[]> {
+    const extractId = (credential: VerifiableCredential<CredentialSubject>) => {
+      const isUuid = credential.id.search('urn:uuid:') > -1;
+      if (isUuid) {
+        return credential.id.split('urn:uuid:').pop() || v4();
+      }
+      return v4();
+    };
+
+    const messages = await Promise.all(
+      credentials.map(async (credential) => {
+        const data = Buffer.from(JSON.stringify(credential), 'utf8').toString(
+          'base64'
+        );
+
+        const message: WebNodeWriteMessage = {
+          data,
+          descriptor: {
+            method: 'CollectionsWrite',
+            objectId: extractId(credential),
+            schema: 'https://www.w3.org/2018/credentials#VerifiableCredential',
+            dateCreated: new Date(credential.issuanceDate).getTime(),
+            datePublished: Date.now(),
+            dataFormat: 'application/ld+json',
+            cid: await this.generateCid(credential),
+          },
+        };
+
+        return message;
+      })
+    );
+
+    const request: WebNodeRequestObject = {
+      requestId: v4(),
+      target: 'vp',
+      messages,
+    };
+
+    const { data: response } = await this._http.post<
+      WebNodeResponseObject<[WebNodeWriteMessage]>
+    >(`/`, request, {
+      baseURL: 'webnode',
+    });
```

Comment:
How does this work with the `ssi-hub` authentication. I think it might work if using browser cookies, but the I think the axios instance in `CacheClient` potentially has an authentication token which it is sending as a Bearer Token. So maybe this call should be in the `CacheClient`?

Reply:
Ah sorry, I missed that `CacheClient` is coming in via the constructor. I think this is fine then

Reply:
We are using `axios` instance from `CacheClient`, so `CacheClient` is taking care of authentication. It's including cookies and headers to each request including to this from `webnode`